superset/common/query_object.py

```diff
@@ -107,8 +115,16 @@ def __init__(
         columns: Optional[List[str]] = None,
         orderby: Optional[List[List[str]]] = None,
         post_processing: Optional[List[Optional[Dict[str, Any]]]] = None,
+        is_rowcount: bool = False,
         **kwargs: Any,
     ):
+        self.is_rowcount = is_rowcount
+        self.datasource = None
+        if datasource:
+            self.datasource = ConnectorRegistry.get_datasource(
+                str(datasource["type"]), int(datasource["id"]), db.session
+            )
```

Comment:
This input does not seem validated at this point. What happens if the `datasource["id"]` does not exist?

Reply:
Perfect opportunity to introduce `TypedDict`